include/onnxruntime/core/session/onnxruntime_cxx_api.h

```diff
@@ -1508,6 +1512,49 @@ struct KernelContext {
   OrtKernelContext* ctx_;
 };
 
+// Logs a message using the provider Ort::Logger, which must be passed by value or reference.
+// Does not throw exceptions, so can be used in destructors and catch clauses.
+// Example: ORT_CXX_LOG(logger, ORT_LOGGING_LEVEL_INFO, "Log a message");
+#define ORT_CXX_LOG(logger, severity, message) \
+  std::ignore = logger.LogMessage(severity, __FILE__, __LINE__, static_cast<const char*>(__FUNCTION__), message)
```

Comment:
>std::ignore [](http://example.com/codeflow?start=2&length=11)

The issue with `std::ignore` is that it is from C++17 and while we require C++17 to build ORT, we do not require users to have C++17 compiler. This was the case last time I checked.

Reply:
According to https://en.cppreference.com/w/cpp/utility/tuple/ignore, `std::ignore` has been available since C++ 11. That said, maybe we just do `static_cast<void>()` to avoid including the tuple header.